python/lsst/afw/typehandling/_GenericMap.cc

```diff
@@ -66,26 +51,48 @@ template <typename K>
 class Publicist : public MutableGenericMap<K> {
 public:
     using typename GenericMap<K>::ConstValueReference;
+    using typename GenericMap<K>::StorableType;
     using GenericMap<K>::unsafeLookup;
     using MutableGenericMap<K>::unsafeErase;
 };
 
+// A combination std::variant visitor and helper function (apply()) that
+// invokes it, to implement item lookup for Python.
+//
+// We copy/clone because there's no good way to prevent other ++ calls from
+// destroying the any reference we might give to Python.
+//
+// Pybind11 does have its own automatic caster for std::variant, but it doesn't
+// handle reference_wrapper types within it (let alone polymorphic Storables),
+// and since we need a visitor to deal with that, there's no point in going
+// back to std::variant instead of straight to Python.
```

Comment:
Is this actually any better than using the custom caster that was removed? It looks a lot more complicated.